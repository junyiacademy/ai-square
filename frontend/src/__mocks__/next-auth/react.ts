export const useSession = jest.fn(() => ({
  data: null,
  status: 'unauthenticated'
}));

export const signIn = jest.fn();
export const signOut = jest.fn();
export const getCsrfToken = jest.fn();
export const getSession = jest.fn();
export const getProviders = jest.fn();

export const SessionProvider = ({ children }: { children: React.ReactNode }) => children;