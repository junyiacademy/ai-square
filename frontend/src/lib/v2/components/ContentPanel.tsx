'use client';

import { Task } from '@/lib/v2/types';
import { BookOpen, FileText, Video, Link as LinkIcon } from 'lucide-react';

interface ContentPanelProps {
  task: Task;
  learningType: string;
  language: string;
}

export function ContentPanel({ task, learningType, language }: ContentPanelProps) {
  const getTaskTypeLabel = () => {
    switch (task.task_type) {
      case 'learning':
        return 'Learning Module';
      case 'practice':
        return 'Practice Activity';
      case 'assessment':
        return 'Assessment';
      default:
        return 'Task';
    }
  };

  const getTaskTypeIcon = () => {
    switch (task.task_type) {
      case 'learning':
        return <BookOpen className="w-5 h-5" />;
      case 'practice':
        return <FileText className="w-5 h-5" />;
      case 'assessment':
        return <FileText className="w-5 h-5" />;
      default:
        return <FileText className="w-5 h-5" />;
    }
  };

  const renderResource = (resource: any) => {
    const icons: Record<string, any> = {
      video: <Video className="w-4 h-4" />,
      article: <FileText className="w-4 h-4" />,
      link: <LinkIcon className="w-4 h-4" />
    };

    return (
      <a
        key={resource.title}
        href={resource.url}
        className="flex items-center gap-2 p-3 bg-gray-50 rounded-lg hover:bg-gray-100 transition-colors"
        target="_blank"
        rel="noopener noreferrer"
      >
        {icons[resource.type] || <LinkIcon className="w-4 h-4" />}
        <span className="text-sm">{resource.title}</span>
      </a>
    );
  };

  return (
    <div className="h-full p-6">
      {/* Task Header */}
      <div className="mb-6">
        <div className="flex items-center gap-2 text-sm text-gray-600 mb-2">
          {getTaskTypeIcon()}
          <span>{getTaskTypeLabel()}</span>
          {task.estimated_minutes && (
            <>
              <span>•</span>
              <span>~{task.estimated_minutes} minutes</span>
            </>
          )}
        </div>
        <h1 className="text-2xl font-bold text-gray-900">{task.title}</h1>
        <p className="text-gray-600 mt-2">{task.description}</p>
      </div>

      {/* Instructions */}
      <div className="mb-6">
        <h2 className="text-lg font-semibold mb-3">Instructions</h2>
        <div className="prose prose-sm max-w-none">
          <p className="text-gray-700 whitespace-pre-wrap">{task.instructions}</p>
        </div>
      </div>

      {/* Learning Objectives (if available) */}
      {task.metadata?.learning_objectives && (
        <div className="mb-6">
          <h2 className="text-lg font-semibold mb-3">Learning Objectives</h2>
          <ul className="space-y-2">
            {task.metadata.learning_objectives.map((objective: string, index: number) => (
              <li key={index} className="flex items-start gap-2">
                <span className="text-blue-500 mt-0.5">•</span>
                <span className="text-gray-700">{objective}</span>
              </li>
            ))}
          </ul>
        </div>
      )}

      {/* Practice Steps (if available) */}
      {task.metadata?.practice_steps && (
        <div className="mb-6">
          <h2 className="text-lg font-semibold mb-3">Practice Steps</h2>
          <ol className="space-y-3">
            {task.metadata.practice_steps.map((step: string, index: number) => (
              <li key={index} className="flex gap-3">
                <span className="flex-shrink-0 w-6 h-6 bg-blue-100 text-blue-600 rounded-full flex items-center justify-center text-sm font-medium">
                  {index + 1}
                </span>
                <span className="text-gray-700">{step}</span>
              </li>
            ))}
          </ol>
        </div>
      )}

      {/* Resources (if available) */}
      {task.metadata?.resources && task.metadata.resources.length > 0 && (
        <div className="mb-6">
          <h2 className="text-lg font-semibold mb-3">Resources</h2>
          <div className="space-y-2">
            {task.metadata.resources.map(renderResource)}
          </div>
        </div>
      )}

      {/* Task-specific content based on type */}
      {task.task_type === 'assessment' && task.metadata?.questions && (
        <div className="mb-6">
          <h2 className="text-lg font-semibold mb-3">Assessment Questions</h2>
          <div className="bg-yellow-50 border border-yellow-200 rounded-lg p-4">
            <p className="text-sm text-yellow-800">
              Complete the questions in the action panel on the right.
              {task.metadata.time_limit && (
                <span className="block mt-1">
                  Time limit: {task.metadata.time_limit} minutes
                </span>
              )}
            </p>
          </div>
        </div>
      )}

      {/* Additional Metadata Display */}
      {task.metadata?.ai_modules && (
        <div className="mt-8 pt-6 border-t">
          <h3 className="text-sm font-medium text-gray-600 mb-2">AI Assistance Available</h3>
          <div className="flex flex-wrap gap-2">
            {task.metadata.ai_modules.map((module: string) => (
              <span
                key={module}
                className="px-2 py-1 bg-blue-100 text-blue-700 rounded text-xs"
              >
                {module.replace(/_/g, ' ')}
              </span>
            ))}
          </div>
        </div>
      )}
    </div>
  );
}