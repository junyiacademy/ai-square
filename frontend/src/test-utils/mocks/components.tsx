/**
 * Common Component Mocks for Testing
 * Centralized mocks for frequently used components and libraries
 */
import React from 'react';

// Framer Motion Mock
export const motionMock = {
  div: ({ children, className, style, ...props }: any) => (
    <div className={className} style={style} {...props}>{children}</div>
  ),
  span: ({ children, className, style, ...props }: any) => (
    <span className={className} style={style} {...props}>{children}</span>
  ),
  button: ({ children, className, style, ...props }: any) => (
    <button className={className} style={style} {...props}>{children}</button>
  ),
  section: ({ children, className, style, ...props }: any) => (
    <section className={className} style={style} {...props}>{children}</section>
  ),
  article: ({ children, className, style, ...props }: any) => (
    <article className={className} style={style} {...props}>{children}</article>
  )
};

// Hero Icons Mock - Outline
export const heroIconsOutlineMock = {
  TrophyIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="trophy-icon"><path /></svg>
  ),
  SparklesIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="sparkles-icon"><path /></svg>
  ),
  CpuChipIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="cpu-chip-icon"><path /></svg>
  ),
  PuzzlePieceIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="puzzle-piece-icon"><path /></svg>
  ),
  GlobeAltIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="globe-alt-icon"><path /></svg>
  ),
  AcademicCapIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="academic-cap-icon"><path /></svg>
  ),
  StarIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="star-icon"><path /></svg>
  ),
  ChartBarIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="chart-bar-icon"><path /></svg>
  ),
  UserIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="user-icon"><path /></svg>
  ),
  ClockIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="clock-icon"><path /></svg>
  ),
  CheckIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="check-icon"><path /></svg>
  ),
  XMarkIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="x-mark-icon"><path /></svg>
  ),
  ArrowLeftIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="arrow-left-icon"><path /></svg>
  ),
  ArrowRightIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="arrow-right-icon"><path /></svg>
  ),
  HomeIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="home-icon"><path /></svg>
  ),
  Cog6ToothIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="cog-6-tooth-icon"><path /></svg>
  )
};

// Hero Icons Mock - Solid
export const heroIconsSolidMock = {
  TrophyIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="trophy-icon-solid"><path /></svg>
  ),
  SparklesIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="sparkles-icon-solid"><path /></svg>
  ),
  CpuChipIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="cpu-chip-icon-solid"><path /></svg>
  ),
  PuzzlePieceIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="puzzle-piece-icon-solid"><path /></svg>
  ),
  GlobeAltIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="globe-alt-icon-solid"><path /></svg>
  ),
  StarIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="star-icon-solid"><path /></svg>
  ),
  ChartBarIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="chart-bar-icon-solid"><path /></svg>
  ),
  UserIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="user-icon-solid"><path /></svg>
  ),
  CheckIcon: ({ className }: { className?: string }) => (
    <svg className={className} data-testid="check-icon-solid"><path /></svg>
  )
};

// React Markdown Mock
export const reactMarkdownMock = ({ children }: { children: string }) => (
  <div data-testid="react-markdown">{children}</div>
);

// Next.js mocks
export const nextImageMock = ({ 
  src, 
  alt, 
  width, 
  height, 
  className, 
  ...props 
}: any) => (
  // eslint-disable-next-line @next/next/no-img-element
  <img 
    src={src} 
    alt={alt} 
    width={width} 
    height={height} 
    className={className}
    data-testid="next-image"
    {...props} 
  />
);

export const nextLinkMock = ({ 
  href, 
  children, 
  className, 
  ...props 
}: any) => (
  <a 
    href={href} 
    className={className}
    data-testid="next-link"
    {...props}
  >
    {children}
  </a>
);

// Common mock setup functions
export const setupFramerMotionMocks = () => {
  jest.mock('framer-motion', () => ({
    motion: motionMock,
    AnimatePresence: ({ children }: any) => children,
    useAnimation: () => ({
      start: jest.fn(),
      stop: jest.fn(),
      set: jest.fn()
    })
  }));
};

export const setupLucideIconsMocks = () => {
  jest.mock('lucide-react', () => ({
    // Common icons used in Discovery components
    Trophy: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="trophy-icon"><path /></svg>
    ),
    Sparkles: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="sparkles-icon"><path /></svg>
    ),
    Cpu: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="cpu-icon"><path /></svg>
    ),
    GraduationCap: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="graduation-cap-icon"><path /></svg>
    ),
    Globe: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="globe-icon"><path /></svg>
    ),
    BarChart: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="bar-chart-icon"><path /></svg>
    ),
    Rocket: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="rocket-icon"><path /></svg>
    ),
    Star: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="star-icon"><path /></svg>
    ),
    // Add more as needed
    ChevronLeft: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="chevron-left-icon"><path /></svg>
    ),
    ChevronRight: ({ className }: { className?: string }) => (
      <svg className={className} data-testid="chevron-right-icon"><path /></svg>
    ),
  }));
};

export const setupNextMocks = () => {
  jest.mock('next/image', () => nextImageMock);
  jest.mock('next/link', () => nextLinkMock);
  
  // Mock Next.js router
  jest.mock('next/router', () => ({
    useRouter: () => ({
      push: jest.fn(),
      replace: jest.fn(),
      pathname: '/',
      query: {},
      asPath: '/',
      back: jest.fn(),
      forward: jest.fn(),
      reload: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn()
      }
    })
  }));
  
  // Mock Next.js navigation
  jest.mock('next/navigation', () => ({
    useRouter: () => ({
      push: jest.fn(),
      replace: jest.fn(),
      back: jest.fn(),
      forward: jest.fn(),
      refresh: jest.fn()
    }),
    useSearchParams: () => new URLSearchParams(),
    usePathname: () => '/'
  }));
};

export const setupReactMarkdownMocks = () => {
  jest.mock('react-markdown', () => reactMarkdownMock);
};