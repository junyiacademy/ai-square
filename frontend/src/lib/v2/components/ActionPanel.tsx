'use client';

import { useState, useEffect } from 'react';
import { Task } from '@/lib/v2/types';
import { Send, CheckCircle, AlertCircle, MessageSquare } from 'lucide-react';

interface ActionPanelProps {
  task: Task;
  learningType: string;
  onComplete: (evaluation: any) => void;
  onAction: (action: string, data: any) => void;
}

export function ActionPanel({ task, learningType, onComplete, onAction }: ActionPanelProps) {
  const [input, setInput] = useState('');
  const [messages, setMessages] = useState<Array<{ role: 'user' | 'assistant'; content: string }>>([]);
  const [selectedAnswers, setSelectedAnswers] = useState<Record<string, any>>({});
  const [isCompleted, setIsCompleted] = useState(false);
  const [evaluation, setEvaluation] = useState<any>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    // Reset state when task changes
    setInput('');
    setMessages([]);
    setSelectedAnswers({});
    setIsCompleted(false);
    setEvaluation(null);
  }, [task.id]);

  const handleSendMessage = async () => {
    if (!input.trim()) return;

    const userMessage = input.trim();
    setInput('');
    setMessages(prev => [...prev, { role: 'user', content: userMessage }]);
    
    onAction('message_sent', { content: userMessage });
    setLoading(true);

    try {
      // Simulate AI response (in real implementation, call API)
      setTimeout(() => {
        const aiResponse = generateAIResponse(task, userMessage);
        setMessages(prev => [...prev, { role: 'assistant', content: aiResponse }]);
        setLoading(false);
      }, 1000);
    } catch (error) {
      setLoading(false);
    }
  };

  const handleAnswerSelect = (questionId: string, answer: any) => {
    setSelectedAnswers(prev => ({
      ...prev,
      [questionId]: answer
    }));
    onAction('answer_selected', { question_id: questionId, answer });
  };

  const handleSubmitAssessment = () => {
    const results = evaluateAssessment(task, selectedAnswers);
    setEvaluation(results);
    setIsCompleted(true);
    onComplete(results);
  };

  const handleTaskComplete = () => {
    const taskEvaluation = {
      completed: true,
      score: 100,
      feedback: 'Great job completing this task!'
    };
    setEvaluation(taskEvaluation);
    setIsCompleted(true);
    onComplete(taskEvaluation);
  };

  // Render different interfaces based on task type
  if (task.task_type === 'assessment' && task.metadata?.questions) {
    return (
      <div className="h-full flex flex-col p-4">
        <h2 className="text-lg font-semibold mb-4">Assessment</h2>
        
        {!isCompleted ? (
          <>
            <div className="flex-1 overflow-y-auto space-y-4 mb-4">
              {task.metadata.questions.map((question: any, index: number) => (
                <div key={question.id} className="bg-white rounded-lg p-4 border">
                  <p className="font-medium mb-3">
                    {index + 1}. {question.question}
                  </p>
                  
                  {question.type === 'multiple_choice' && (
                    <div className="space-y-2">
                      {question.options.map((option: string, optIndex: number) => (
                        <label
                          key={optIndex}
                          className="flex items-center gap-2 p-2 rounded hover:bg-gray-50 cursor-pointer"
                        >
                          <input
                            type="radio"
                            name={question.id}
                            value={optIndex}
                            checked={selectedAnswers[question.id] === optIndex}
                            onChange={() => handleAnswerSelect(question.id, optIndex)}
                            className="text-blue-600"
                          />
                          <span>{option}</span>
                        </label>
                      ))}
                    </div>
                  )}
                </div>
              ))}
            </div>
            
            <button
              onClick={handleSubmitAssessment}
              disabled={Object.keys(selectedAnswers).length < task.metadata.questions.length}
              className="w-full px-4 py-2 bg-blue-600 text-white rounded-lg hover:bg-blue-700 disabled:opacity-50 disabled:cursor-not-allowed"
            >
              Submit Assessment
            </button>
          </>
        ) : (
          <div className="flex-1 flex flex-col items-center justify-center">
            <CheckCircle className="w-16 h-16 text-green-500 mb-4" />
            <h3 className="text-xl font-semibold mb-2">Assessment Complete!</h3>
            <p className="text-gray-600 text-center mb-4">
              Score: {evaluation?.score}%
            </p>
            <p className="text-gray-600 text-center">
              {evaluation?.feedback}
            </p>
          </div>
        )}
      </div>
    );
  }

  // Default chat interface for learning and practice
  return (
    <div className="h-full flex flex-col p-4">
      <h2 className="text-lg font-semibold mb-4 flex items-center gap-2">
        <MessageSquare className="w-5 h-5" />
        AI Assistant
      </h2>
      
      {!isCompleted ? (
        <>
          {/* Chat Messages */}
          <div className="flex-1 overflow-y-auto mb-4 space-y-3">
            {messages.length === 0 && (
              <div className="text-center text-gray-500 py-8">
                <p>Start a conversation with your AI tutor!</p>
                <p className="text-sm mt-2">Ask questions or request help with the task.</p>
              </div>
            )}
            
            {messages.map((message, index) => (
              <div
                key={index}
                className={`flex ${message.role === 'user' ? 'justify-end' : 'justify-start'}`}
              >
                <div
                  className={`max-w-[80%] rounded-lg p-3 ${
                    message.role === 'user'
                      ? 'bg-blue-600 text-white'
                      : 'bg-gray-100 text-gray-800'
                  }`}
                >
                  <p className="text-sm">{message.content}</p>
                </div>
              </div>
            ))}
            
            {loading && (
              <div className="flex justify-start">
                <div className="bg-gray-100 rounded-lg p-3">
                  <div className="flex gap-1">
                    <div className="w-2 h-2 bg-gray-400 rounded-full animate-bounce" />
                    <div className="w-2 h-2 bg-gray-400 rounded-full animate-bounce delay-100" />
                    <div className="w-2 h-2 bg-gray-400 rounded-full animate-bounce delay-200" />
                  </div>
                </div>
              </div>
            )}
          </div>
          
          {/* Input Area */}
          <div className="space-y-3">
            <div className="flex gap-2">
              <input
                type="text"
                value={input}
                onChange={(e) => setInput(e.target.value)}
                onKeyPress={(e) => e.key === 'Enter' && handleSendMessage()}
                placeholder="Type your message..."
                className="flex-1 px-3 py-2 border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
              />
              <button
                onClick={handleSendMessage}
                disabled={!input.trim() || loading}
                className="px-4 py-2 bg-blue-600 text-white rounded-lg hover:bg-blue-700 disabled:opacity-50"
              >
                <Send className="w-4 h-4" />
              </button>
            </div>
            
            {task.task_type !== 'assessment' && (
              <button
                onClick={handleTaskComplete}
                className="w-full px-4 py-2 bg-green-600 text-white rounded-lg hover:bg-green-700"
              >
                Mark Task as Complete
              </button>
            )}
          </div>
        </>
      ) : (
        <div className="flex-1 flex flex-col items-center justify-center">
          <CheckCircle className="w-16 h-16 text-green-500 mb-4" />
          <h3 className="text-xl font-semibold mb-2">Task Complete!</h3>
          <p className="text-gray-600 text-center">
            {evaluation?.feedback || 'Great job! You can now proceed to the next task.'}
          </p>
        </div>
      )}
    </div>
  );
}

// Helper functions
function generateAIResponse(task: Task, userMessage: string): string {
  // In real implementation, this would call an AI API
  const responses = [
    "That's a great question! Let me help you understand this better...",
    "You're on the right track! Consider thinking about...",
    "Excellent observation! This relates to...",
    "Let me clarify that concept for you..."
  ];
  
  return responses[Math.floor(Math.random() * responses.length)] + 
    ` Based on the task "${task.title}", ${userMessage.toLowerCase()}`;
}

function evaluateAssessment(task: Task, answers: Record<string, any>) {
  if (!task.metadata?.questions) return null;
  
  let correct = 0;
  const total = task.metadata.questions.length;
  
  task.metadata.questions.forEach((question: any) => {
    if (answers[question.id] === question.correct_answer) {
      correct++;
    }
  });
  
  const score = Math.round((correct / total) * 100);
  
  return {
    score,
    correct,
    total,
    feedback: score >= 70 
      ? 'Excellent work! You have a strong understanding of the concepts.'
      : 'Good effort! Review the material and try again to improve your score.'
  };
}