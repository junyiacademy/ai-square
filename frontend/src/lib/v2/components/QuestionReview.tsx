'use client';

import { useState } from 'react';
import { ChevronLeft, ChevronRight, X, CheckCircle, XCircle } from 'lucide-react';

interface Question {
  id: string;
  question: string;
  options?: {
    a: string;
    b: string;
    c: string;
    d: string;
  };
  correct_answer?: string;
  explanation?: string;
  type?: string;
}

interface UserAnswer {
  questionId: string;
  answer: string;
  isCorrect: boolean;
  timestamp?: Date;
}

interface QuestionReviewProps {
  questions: Question[];
  userAnswers: UserAnswer[];
  selectedQuestionIds: string[];
  onClose: () => void;
  title?: string;
}

export function QuestionReview({ 
  questions, 
  userAnswers, 
  selectedQuestionIds,
  onClose,
  title = "Related Questions"
}: QuestionReviewProps) {
  const [currentIndex, setCurrentIndex] = useState(0);

  // Filter questions based on selected IDs
  const selectedQuestions = questions.filter(q => selectedQuestionIds.includes(q.id));
  
  if (selectedQuestions.length === 0) {
    return (
      <div className="bg-white rounded-lg border border-gray-200 p-6">
        <div className="flex items-center justify-between mb-4">
          <h3 className="text-lg font-semibold text-gray-900">{title}</h3>
          <button
            onClick={onClose}
            className="text-gray-400 hover:text-gray-600"
          >
            <X className="w-5 h-5" />
          </button>
        </div>
        <div className="text-center text-gray-500 py-8">
          No questions available for this node
        </div>
      </div>
    );
  }

  const currentQuestion = selectedQuestions[currentIndex];
  const userAnswer = userAnswers.find(a => a.questionId === currentQuestion.id);

  const handleNext = () => {
    if (currentIndex < selectedQuestions.length - 1) {
      setCurrentIndex(currentIndex + 1);
    }
  };

  const handlePrevious = () => {
    if (currentIndex > 0) {
      setCurrentIndex(currentIndex - 1);
    }
  };

  const getOptionClass = (optionKey: string) => {
    if (!userAnswer) return 'border-gray-300 hover:bg-gray-50';
    
    const isUserChoice = userAnswer.answer === optionKey;
    const isCorrect = currentQuestion.correct_answer === optionKey;
    
    if (isUserChoice && isCorrect) {
      return 'border-green-500 bg-green-50 text-green-900'; // User chose correct
    } else if (isUserChoice && !isCorrect) {
      return 'border-red-500 bg-red-50 text-red-900'; // User chose wrong
    } else if (!isUserChoice && isCorrect) {
      return 'border-green-300 bg-green-25 text-green-700'; // Correct answer not chosen
    }
    return 'border-gray-300'; // Other options
  };

  const getOptionIcon = (optionKey: string) => {
    if (!userAnswer) return null;
    
    const isUserChoice = userAnswer.answer === optionKey;
    const isCorrect = currentQuestion.correct_answer === optionKey;
    
    if (isUserChoice && isCorrect) {
      return <CheckCircle className="w-5 h-5 text-green-600" />;
    } else if (isUserChoice && !isCorrect) {
      return <XCircle className="w-5 h-5 text-red-600" />;
    } else if (!isUserChoice && isCorrect) {
      return <CheckCircle className="w-5 h-5 text-green-500" />;
    }
    return null;
  };

  return (
    <div className="bg-white rounded-lg border border-gray-200 overflow-hidden">
      {/* Header */}
      <div className="bg-gray-50 px-6 py-4 border-b">
        <div className="flex items-center justify-between">
          <h3 className="text-lg font-semibold text-gray-900">{title}</h3>
          <button
            onClick={onClose}
            className="text-gray-400 hover:text-gray-600 transition-colors"
          >
            <X className="w-5 h-5" />
          </button>
        </div>
        
        {/* Question Navigation */}
        <div className="flex items-center justify-between mt-3">
          <div className="text-sm text-gray-600">
            Question {currentIndex + 1} of {selectedQuestions.length}
          </div>
          <div className="flex items-center gap-2">
            <button
              onClick={handlePrevious}
              disabled={currentIndex === 0}
              className={`p-2 rounded-lg transition-colors ${
                currentIndex === 0
                  ? 'text-gray-300 cursor-not-allowed'
                  : 'text-gray-600 hover:text-gray-900 hover:bg-gray-100'
              }`}
            >
              <ChevronLeft className="w-4 h-4" />
            </button>
            <button
              onClick={handleNext}
              disabled={currentIndex === selectedQuestions.length - 1}
              className={`p-2 rounded-lg transition-colors ${
                currentIndex === selectedQuestions.length - 1
                  ? 'text-gray-300 cursor-not-allowed'
                  : 'text-gray-600 hover:text-gray-900 hover:bg-gray-100'
              }`}
            >
              <ChevronRight className="w-4 h-4" />
            </button>
          </div>
        </div>
      </div>

      {/* Question Content */}
      <div className="p-6">
        {/* Question */}
        <div className="mb-6">
          <h4 className="text-base font-medium text-gray-900 mb-4 leading-relaxed">
            {currentQuestion.question}
          </h4>
          
          {/* Answer Status */}
          {userAnswer && (
            <div className={`inline-flex items-center gap-2 px-3 py-1 rounded-full text-sm font-medium mb-4 ${
              userAnswer.isCorrect 
                ? 'bg-green-100 text-green-800' 
                : 'bg-red-100 text-red-800'
            }`}>
              {userAnswer.isCorrect ? (
                <>
                  <CheckCircle className="w-4 h-4" />
                  Correct
                </>
              ) : (
                <>
                  <XCircle className="w-4 h-4" />
                  Incorrect
                </>
              )}
            </div>
          )}
        </div>

        {/* Options */}
        {currentQuestion.options && (
          <div className="space-y-3 mb-6">
            {Object.entries(currentQuestion.options).map(([optionKey, optionText]) => (
              <div
                key={optionKey}
                className={`p-4 rounded-lg border-2 transition-colors ${getOptionClass(optionKey)}`}
              >
                <div className="flex items-start gap-3">
                  <div className="flex-shrink-0 w-6 h-6 rounded-full bg-gray-100 text-gray-600 text-sm font-medium flex items-center justify-center mt-0.5">
                    {optionKey.toUpperCase()}
                  </div>
                  <div className="flex-1">
                    <p className="text-sm leading-relaxed">{optionText}</p>
                  </div>
                  <div className="flex-shrink-0">
                    {getOptionIcon(optionKey)}
                  </div>
                </div>
              </div>
            ))}
          </div>
        )}

        {/* Explanation */}
        {currentQuestion.explanation && (
          <div className="bg-blue-50 border border-blue-200 rounded-lg p-4">
            <h5 className="font-medium text-blue-900 mb-2">Explanation</h5>
            <p className="text-sm text-blue-800 leading-relaxed">
              {currentQuestion.explanation}
            </p>
          </div>
        )}
      </div>

      {/* Progress Indicators */}
      <div className="px-6 py-3 bg-gray-50 border-t">
        <div className="flex gap-1">
          {selectedQuestions.map((_, index) => (
            <button
              key={index}
              onClick={() => setCurrentIndex(index)}
              className={`w-3 h-3 rounded-full transition-colors ${
                index === currentIndex
                  ? 'bg-blue-600'
                  : index < currentIndex
                  ? 'bg-blue-300'
                  : 'bg-gray-300'
              }`}
            />
          ))}
        </div>
      </div>
    </div>
  );
}