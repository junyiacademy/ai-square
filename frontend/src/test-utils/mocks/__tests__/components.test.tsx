/**
 * Tests for components
 */

import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import components from '../components';

// Mock dependencies
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    back: jest.fn(),
    refresh: jest.fn(),
  }),
  usePathname: () => '/',
  useSearchParams: () => new URLSearchParams(),
}));

describe('components', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render without crashing', () => {
    const { container } = render(<components />);
    expect(container).toBeInTheDocument();
  });

  it('should have proper structure', () => {
    render(<components />);
    
    // Check for basic elements - adjust based on component
    const element = screen.getByRole('main', { hidden: true }) || 
                   screen.getByRole('article', { hidden: true }) ||
                   screen.getByRole('section', { hidden: true }) ||
                   document.querySelector('div');
    expect(element).toBeInTheDocument();
  });

  it('should handle user interactions', async () => {
    render(<components />);
    
    // Look for interactive elements
    const buttons = screen.queryAllByRole('button');
    const links = screen.queryAllByRole('link');
    const inputs = screen.queryAllByRole('textbox');
    
    // Test at least one interaction if available
    if (buttons.length > 0) {
      fireEvent.click(buttons[0]);
      // Add assertion based on expected behavior
    }
    
    if (inputs.length > 0) {
      fireEvent.change(inputs[0], { target: { value: 'test' } });
      expect(inputs[0]).toHaveValue('test');
    }
  });

  it('should be accessible', () => {
    const { container } = render(<components />);
    
    // Basic accessibility checks
    const headings = container.querySelectorAll('h1, h2, h3, h4, h5, h6');
    const images = container.querySelectorAll('img');
    
    images.forEach(img => {
      expect(img).toHaveAttribute('alt');
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<components />);
    expect(container.firstChild).toMatchSnapshot();
  });
});