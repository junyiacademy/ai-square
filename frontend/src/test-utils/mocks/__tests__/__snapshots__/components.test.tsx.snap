// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components should match snapshot 1`] = `<components />`;
